src/ComposerRequireChecker/NodeVisitor/UsedSymbolCollector.php

```diff
@@ -52,130 +57,151 @@ public function enterNode(Node $node)
         return parent::enterNode($node);
     }
 
-    private function recordExtendsUsage(Node $node)
+    private function recordExtendsUsage(Node $node): void
     {
         if ($node instanceof Node\Stmt\Class_) {
             array_map([$this, 'recordUsageOf'], array_filter([$node->extends]));
         }
 
-        if ($node instanceof Node\Stmt\Interface_) {
-            array_map([$this, 'recordUsageOf'], array_filter($node->extends));
+        if (! ($node instanceof Node\Stmt\Interface_)) {
+            return;
         }
+
+        array_map([$this, 'recordUsageOf'], array_filter($node->extends));
     }
 
-    private function recordImplementsUsage(Node $node)
+    private function recordImplementsUsage(Node $node): void
     {
-        if ($node instanceof Node\Stmt\Class_) {
-            array_map([$this, 'recordUsageOf'], $node->implements);
+        if (! ($node instanceof Node\Stmt\Class_)) {
+            return;
         }
+
+        array_map([$this, 'recordUsageOf'], $node->implements);
     }
 
-    private function recordClassExpressionUsage(Node $node)
+    private function recordClassExpressionUsage(Node $node): void
     {
-        if (($node instanceof Node\Expr\StaticCall
-                || $node instanceof Node\Expr\StaticPropertyFetch
-                || $node instanceof Node\Expr\ClassConstFetch
-                || $node instanceof Node\Expr\New_
-                || $node instanceof Node\Expr\Instanceof_
-            )
-            && $node->class instanceof Node\Name
+        if (
+            (! ($node instanceof Node\Expr\StaticCall)
```

Comment:
This conditional got much harder to read: a simpler way to change this is to reverse it with a single `!` in front of the original version

Reply:
With this psalm could not get the type right. Now the condition is split in two and is with this IMHO definitely more readable than before. WDYT?